mitosheet/src/mito/components/SearchBar.tsx

```diff
@@ -7,22 +7,48 @@ import XIcon from './icons/XIcon';
 import '../../../css/elements/SearchBar.css';
 import { classNames } from '../utils/classNames';
 import Input from './elements/Input';
+import { MitoAPI } from '../api/api';
 
 interface SearchBarProps {
     setUIState: React.Dispatch<React.SetStateAction<UIState>>;
     uiState: UIState;
+    mitoAPI: MitoAPI;
 }
 
 export const SearchBar: React.FC<SearchBarProps> = (props) => {
+    const { setUIState, uiState, mitoAPI } = props;
+    const [totalMatches, setTotalMatches] = React.useState<number | null>(null);
+    const searchValue = uiState.currOpenSearch.searchValue;
+
+    /**
+     * Update the total number of matches when the search value changes
+     */
+    React.useEffect(() => {
```

Comment:
I think it would be really nice if we debounced this API call slightly, so when users are typing it doesn't show them out of date results - and also so we do less work on the backend. 

Want to add debouncing to the useStateFromAPIAsync hook? You can look at the useDebouncedHook for implementation ideas!

Reply:
If you're not sure on debouncing: https://www.techtarget.com/whatis/definition/debouncing#:~:text=Debouncing%20is%20removing%20unwanted%20input,hardware%20switches%2C%20programs%20and%20websites.

Reply:
Make sure to test other usages of this hook after making changes (they don't need to be debounced) -- but just make sure they aren't broken. 

Debouncing can be an optional parameter, methinks.